packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -106,125 +33,30 @@ function calculateContent(trace: Trace, span: Span, allSpans: Span[], resultValu
-          const overlayWithout = [];
-          for (let i = 0; i < overlayOnly.length; i++) {
-            if (!earliestLongerAsParent.relativeStartTime <= overlayOnly[i].relativeStartTime) {
-              overlayWithout.push(overlayOnly[i]);
-            }
-          }
-          for (let i = 0; i < overlayWithout.length; i++) {
-            if (
-              overlayWithout[i].relativeStartTime + overlayWithout[i].duration >
-              earliestLongerAsParent.relativeStartTime
-            ) {
-              overlayWithout[i].duration -=
-                overlayWithout[i].relativeStartTime +
-                overlayWithout[i].duration -
-                earliestLongerAsParent.relativeStartTime;
-              if (overlayWithout[i].duration < 0) {
-                overlayWithout[i].duration = 0;
-              }
-            }
-          }
-
-          tempSelf = onlyOverlay(overlayWithout, allChildrenWithout, tempSelf, span);
-          const diff = span.relativeStartTime + span.duration - earliestLongerAsParent.relativeStartTime;
-          tempSelf = Math.max(0, tempSelf - diff);
-        }
-      } else if (longerAsParent) {
-        // span is longer as Parent
-        tempSelf = longerAsParentSpan[0].relativeStartTime - span.relativeStartTime;
-        for (let i = 0; i < allChildren.length; i++) {
-          if (allChildren[i].spanID !== longerAsParentSpan[0].spanID) {
-            tempSelf -= allChildren[i].duration;
-          }
-        }
-      } else {
-        // Overlay
-        tempSelf = onlyOverlay(allOverlay, allChildren, tempSelf, span);
-      }
-    }
-    // no children
+    // DRange treats all intervals as inclusive, even for subtract operations.
+    // We want to represent spans as half-open intervals like [startTime, startTime + duration).
+    // This way the subtraction preserves the right boundaries. However, DRange treats all
+    // intervals as exclusive. For example,
+    //       range(1, 10).subtract(4, 8) => range([1, 3], [9-10])
+    //       length=(3-1)+(10-9)=2+1=3
+    // In other words, we took an interval of length=10-1=9 and subtracted length=8-4=4.
+    // We should've ended up with length 9-4=5, but we got 3.
+    // To work around that, we multiply start/end times by 10 and subtract one from the end.
+    // So instead of [1-10] we get [10-99]. This makes the intervals work like half-open.
+    const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
+    // Filter for CHILD_OF we don't want to calculate FOLLOWS_FROM
+    const children = allSpans.filter(
+      each =>
+        span.childSpanIds.includes(each.spanID) &&
+        each.references[0].spanID === span.spanID &&
+        each.references[0].refType === 'CHILD_OF'
+    );
+    children.forEach(child => {
+      spanRange.subtract(10 * child.startTime, 10 * (child.startTime + child.duration) - 1);
+    });
+    const selfTime = spanRange.length;
+    tempSelf += Math.round(selfTime / 10);
```

Comment:
```suggestion
    tempSelf += Math.round(spanRange.length / 10);
```
Semantically the first line is incorrect. Also, it's not clear to me why this is using `tempSelf +=` - isn't there only one pass through the code? Why accumulation?

I would suggest moving this out to a separate function computeSelfTime.